plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -140,6 +155,51 @@ def associate(source_repo, dest_repo, import_conduit, config, units=None, solver
     return associated_units, failed_units
 
 
+def get_recursive_module_deps(module_units):
+    """
+    Module Metadata specifies dependencies on other modules. This method is useful
+    for taking a set of Modulemd documents and returning the total set of module
+    dependencies for those units.
+
+    :param module_units: iterable of Modulemd
+    :type module_units: iterable
+
+    :return: set of Modulemd
+    :rtype: set
+    """
+    module_deps = set()
+    for unit in module_units:
+        for dependency in unit.dependencies:
+            # we're not a real client, so we don't care about the platform
+            # we need to copy all dependencies for any platform
+            dependency.pop("platform")
+
+            for name, streams in dependency.items():
+                all_module_streams = models.Modulemd.objects.filter(name=name)
+
+                # If there's no stream specified, just copy them all
+                # Better behavior would probably be to copy the default stream, if one exists
+                # TODO?
```

Comment:
I would class this as a low-priority issue on the basis that only one or two modules specify their dependencies as an empty list.  But it's worth mentioning that it means that in those cases, even recursive-conservative will be copying a couple of extra streams.  Address in Pulp 3, probably.

Another concern is the fact that copying a module:stream will copy all of the historical versions of that module:stream.  Streams like "master" and "rolling"  will get new versions on a regular basis, and if those don't get cleaned out regularly with force-full syncs, then we're going to be copying (and thus performing these recursive operations on, as well as depsolving for) a huge number of module documents that will likely never be used.

Reply:
What is the regression? If we use approach #2, only default streams are copied for module dependencies?

Reply:
I think so.  It seemed to do that in a couple of my tests.  But it's hard to know for sure if it always happens correctly.

Reply:
I'll ask what DNF does

Reply:
Ah and you mentioned that it's a rare case when the stream is not specified, so we should be ok.